pkg/plugins/secret/secrets_manager/secrets_manager_test.go

```diff
@@ -0,0 +1,324 @@
+					mockSecretClient.EXPECT().CreateSecret(
+						&secretsmanager.CreateSecretInput{
+							Name:         aws.String("Test"),
+							SecretBinary: testSecretVal,
+						},
+					).Return(&secretsmanager.CreateSecretOutput{
+						ARN:       aws.String(testARN),
+						Name:      aws.String("Test"),
+						VersionId: aws.String(testVersionID),
+					}, nil).Times(1)
+
+					response, err := secretPlugin.Put(&testSecret, testSecretVal)
+					By("Not returning an error")
+					Expect(err).ShouldNot(HaveOccurred())
+					By("Returning a response with a version id")
+					Expect(response.SecretVersion.Version).To(Equal(testVersionID))
+				})
+			})
+			When("Putting an empty secret", func() {
+				secretPlugin := &secretsManagerSecretService{}
+
+				It("Should return an error", func() {
+					var emptySecret = &secret.Secret{}
+					response, err := secretPlugin.Put(emptySecret, testSecretVal)
+					By("Returning an error")
+					Expect(err).Should(HaveOccurred())
+					By("Returning a nil response")
+					Expect(response).Should(BeNil())
+				})
+			})
+			When("Putting a nil secret", func() {
+				secretPlugin := &secretsManagerSecretService{}
+
+				It("Should return an error", func() {
+					response, err := secretPlugin.Put(nil, testSecretVal)
+					By("Returning an error")
+					Expect(err).Should(HaveOccurred())
+					By("Returning a nil response")
+					Expect(response).Should(BeNil())
+				})
+			})
+
+			When("Putting a secret with a nil value", func() {
+				secretPlugin := &secretsManagerSecretService{}
+
+				It("Should return an error", func() {
+					response, err := secretPlugin.Put(&testSecret, nil)
+					By("Returning an error")
+					Expect(err).Should(HaveOccurred())
+					By("Returning a nil response")
+					Expect(response).Should(BeNil())
+				})
+			})
+
+			When("AWS SecretsManager.GetSecretValue returns an non NOT_FOUND error", func() {
+				ctrl := gomock.NewController(GinkgoT())
+				mockSecretClient := mocks.NewMockSecretsManagerAPI(ctrl)
+				secretPlugin := &secretsManagerSecretService{
+					client: mockSecretClient,
+				}
+				It("Should successfully store a secret", func() {
+					defer ctrl.Finish()
+
+					mockSecretClient.EXPECT().GetSecretValue(
+						gomock.Any(),
+					).Return(nil, fmt.Errorf("non-aws error")).Times(1)
+
+					response, err := secretPlugin.Put(&testSecret, testSecretVal)
+					By("returning an error")
+					Expect(err).Should(HaveOccurred())
+					Expect(err.Error()).To(Equal("failed to retrieve secret: non-aws error"))
+					By("returning a nil response")
+					Expect(response).Should(BeNil())
+				})
+			})
+
+			When("AWS SecretsManager.PutSecretValue returns an AWS error", func() {
+				ctrl := gomock.NewController(GinkgoT())
+				mockSecretClient := mocks.NewMockSecretsManagerAPI(ctrl)
+				secretPlugin := &secretsManagerSecretService{
+					client: mockSecretClient,
+				}
+				It("Should successfully store a secret", func() {
```

Comment:
Change `It("..", func() {` message to reflect scenario

Reply:
done